ush/python/pygfs/jedi/jedi.py

```diff
@@ -22,249 +20,260 @@ class Jedi:
-
-        copylist = []
-        for ob in observations['observers']:
-            obfile = ob['obs space']['obsdatain']['engine']['obsfile']
-            basename = os.path.basename(obfile)
-            copylist.append([os.path.join(task_config.COM_OBS, basename), obfile])
-        obs_dict = {
-            'mkdir': [os.path.join(task_config.DATA, 'obs')],
-            'copy': copylist
-        }
-        return obs_dict
+        # TODO: linking is not permitted per EE2.
+        # Needs work in JEDI to be able to copy the exec. [NOAA-EMC/GDASApp#1254]
+        logger.warn("Linking is not permitted per EE2.")
+        if not os.path.exists(self.jedi_config.exe):
+            os.symlink(self.jedi_config.exe_src, self.jedi_config.exe)
 
+    @staticmethod
     @logit(logger)
-    def get_bias_dict(self, task_config: AttrDict, bias_file) -> Dict[str, Any]:
-        """Compile a dictionary of observation files to copy
-
-        This method extracts 'observers' from the JEDI yaml and determines from that list
-        if bias correction tar files are to be copied to the run directory
-        from the component directory.
+    def remove_redundant(input_list: List) -> List:
+        """Remove reduncancies from list with possible redundant, non-mutable elements
 
         Parameters
         ----------
-        task_config: AttrDict
-            Attribute-dictionary of all configuration variables associated with a GDAS task.
-        bias_file
-            name of bias correction tar file
+        input_list : List
+            List with possible redundant, non-mutable elements
 
         Returns
         ----------
-        bias_dict: Dict
-            a dictionary containing the list of observation bias files to copy for FileHandler
+        output_list : List
+            Input list but with redundancies removed
         """
 
-        observations = find_value_in_nested_dict(self.config, 'observations')
-
-        copylist = []
-        for ob in observations['observers']:
-            if 'obs bias' in ob.keys():
-                obfile = ob['obs bias']['input file']
-                obdir = os.path.dirname(obfile)
-                basename = os.path.basename(obfile)
-                prefix = '.'.join(basename.split('.')[:-3])
-                bfile = f"{prefix}.{bias_file}"
-                tar_file = os.path.join(obdir, bfile)
-                copylist.append([os.path.join(task_config.VarBcDir, bfile), tar_file])
-                break
-
-        bias_dict = {
-            'mkdir': [os.path.join(task_config.DATA, 'bc')],
-            'copy': copylist
-        }
+        output_list = []
+        for item in input_list:
+            if item not in output_list:
+                output_list.append(item)
 
-        return bias_dict
+        return output_list
 
     @staticmethod
     @logit(logger)
-    def extract_tar(tar_file: str) -> None:
-        """Extract files from a tarball
+    def extract_tar_from_fh_dict(fh_dict) -> None:
```

Comment:
picking nits, but `fh` to me means "forecast hour" at first glance, perhaps `file_dict`?